slatedb/src/tablestore.rs

```diff
@@ -516,6 +503,48 @@ impl TableStore {
     }
 }
 
+async fn write_sst_with_retry(
```

Comment:
Might want to consider generalizing this so we can use it for manifest (manifest_storel.rs) and compactor files (forthcoming), too.

Reply:
Also, take a look at wal_buffer.rs:

```rs
// not all the flush error is fatal. on temporary network errors, we can retry later.
// After a few continuous failures, we'll return the error, and set it as fatal error
// in cleanup.
match result {
    Ok(_) => {
        contiguous_failures_count = 0;
    }
    Err(e) => {
        contiguous_failures_count += 1;
        if contiguous_failures_count > 3 {
            return Err(e);
        }
    }
}
```